File: caisse.cpp
#include "caisse.h"

Caisse:: Caisse(int x, int y, bool Etat, int m_type): Coordonnees(x,y)
{
    EtatCaisse = Etat;
    type = m_type;
}


bool Caisse::getEtat(){
    return EtatCaisse;
}


void Caisse::ChangementCouleur(){
    // changer la photo de la caisse
    EtatCaisse = !EtatCaisse;
}


void Caisse:: afficher_caisse(QPainter *p){
    if (EtatCaisse==true)
        p->drawPixmap(getY()*50,getX()*50+25,50,50, QPixmap(":/images/data/PNG/caisse - marque.png"));
    else
        p->drawPixmap(getY()*50,getX()*50+25,50,50, QPixmap(":/images/data/PNG/caisse.png"));

}

File: Caisse.java
package code;

import javax.swing.ImageIcon;
import java.awt.Graphics;

public class Caisse extends Coordonnees{
	
	//Attributs
	protected boolean EtatCaisse=false;
	protected int type=4;
	public ImageIcon ImCaisse;

	public Caisse(int x, int y,  boolean Etat,int m_type) {
		super(x,y);
		EtatCaisse = Etat;
		type = m_type;
		if (EtatCaisse == true) 
			ImCaisse = new ImageIcon("PNG/caisse-marque.png");
		else
			ImCaisse = new ImageIcon("PNG/caisse.png");
	}
	
	public boolean getEtat() {return EtatCaisse;}
	
	//public Coordonnees getPosition() {return super(this.getI(),this.getJ());}
	
	public void ChangementCouleur() {EtatCaisse = !EtatCaisse;}
	
	public void afficher_caisse(Graphics g) {
		if (EtatCaisse == true) 
			ImCaisse = new ImageIcon("PNG/caisse-marque.png");
		else
			ImCaisse = new ImageIcon("PNG/caisse.png");
		g.drawImage(this.ImCaisse.getImage(),this.getJ()*50,this.getI()*50+25, 50,50, null);
	}
}
